feature/lldp/otg_tests/core_lldp_tlv_population_test/core_lldp_tlv_population_test.go

```diff
@@ -214,56 +207,46 @@ func verifyNodeConfig(t *testing.T, node gnmi.DeviceOrOpts, port *ondatra.Port,
 	}
 }
 
-func waitFor(fn func() bool, t testing.TB) {
-	start := time.Now()
-	for {
-		done := fn()
-		if done {
-			t.Logf("Expected LLDP Stats/Metric received")
-			break
-		}
-		if time.Since(start) > 65*time.Second {
-			t.Error("Timeout while waiting for expected stats...")
-		}
-		time.Sleep(500 * time.Millisecond)
-	}
-}
-
 // checkOtgLldpMetrics verifies OTG side lldp Metrics values based on DUT side lldp is enabled or not
-func checkOtgLldpMetrics(t *testing.T, otg *otg.OTG, c gosnappi.Config, lldpEnabled bool) bool {
-	otgutils.LogLLDPMetrics(t, otg, c)
+func checkOtgLldpMetrics(t *testing.T, otg *otg.OTG, c gosnappi.Config, lldpEnabled bool) {
 	for _, lldp := range c.Lldp().Items() {
-		lldpName := lldp.Name()
-		lldpState := gnmi.Get(t, otg, gnmi.OTG().LldpInterface(lldpName).State())
-		lldpStateCounters := lldpState.GetCounters()
-
-		if lldpStateCounters.GetFrameOut() == 0 {
-			return false
+		lastValue, ok := gnmi.Watch(t, otg, gnmi.OTG().LldpInterface(lldp.Name()).Counters().FrameOut().State(), time.Minute, func(v *ygnmi.Value[uint64]) bool {
+			time.Sleep(1 * time.Second)
+			txPackets, _ := v.Val()
+			return v.IsPresent() && txPackets != 0
+		}).Await(t)
+		if !ok {
+			txPackets, _ := lastValue.Val()
+			t.Errorf("LLDP sent packets got: %v, want: > 0.", txPackets)
 		}
-
+		framesIn, _ := gnmi.Watch(t, otg, gnmi.OTG().LldpInterface(lldp.Name()).Counters().FrameIn().State(), time.Minute, func(v *ygnmi.Value[uint64]) bool {
+			time.Sleep(1 * time.Second)
+			return v.IsPresent()
+		}).Await(t)
+		otgutils.LogLLDPMetrics(t, otg, c)
 		if lldpEnabled {
-			if lldpStateCounters.GetFrameIn() == 0 {
-				return false
+			if rxPackets, _ := framesIn.Val(); rxPackets == 0 {
+				t.Errorf("LLDP received packets got: %v, want: > 0.", rxPackets)
 			}
 		} else {
-			if lldpStateCounters.GetFrameIn() != 0 {
-				return false
+			if rxPackets, _ := framesIn.Val(); rxPackets != 0 {
+				t.Errorf("LLDP received packets got: %v, want: 0.", rxPackets)
 			}
 		}
 	}
-	return true
 }
 
 // checkOtgLldpMetrics verifies OTG side lldp neighbor states
-func checkOtgLldpNeighbors(t *testing.T, otg *otg.OTG, c gosnappi.Config, expLldpNeighbors map[string][]lldpNeighbors) bool {
-	// otgutils.LogLLDPNeighborStates(t, otg, c)
+func checkOtgLldpNeighbors(t *testing.T, otg *otg.OTG, c gosnappi.Config, expLldpNeighbors map[string][]lldpNeighbors) {
+	otgutils.LogLLDPNeighborStates(t, otg, c)
 
 	for lldp, lldpNeighbors := range expLldpNeighbors {
 		lldpState := gnmi.Get(t, otg, gnmi.OTG().LldpInterface(lldp).State())
 
 		if len(lldpNeighbors) == 0 {
 			if lldpState.LldpNeighborDatabase != nil {
-				return false
+				t.Errorf("LLDP Neighbordatabase not expected nil")
```

Comment:
Can we include LldpNeighborDatabase in the error message if it's not nil? If that happens, we'd like to quickly see more details.

Something like: t.Errorf("LldpNeighborDatabase = %v; want nil", lldpState.LldpNeighborDatabase)

https://github.com/golang/go/wiki/CodeReviewComments#useful-test-failures